compiler/ast_to_ir.py

```diff
@@ -132,7 +133,7 @@ def combine_pipe(ast_nodes):
     else:
         ## If any part of the pipe is not an IR, the compilation must fail.
         log("Node: {} is not pure".format(ast_nodes[0]))
-        raise Exception('Not pure node in pipe')
+        raise UnparallelizableError('Not pure node in pipe')
```

Comment:
I would include more information here (like the node), even though it is logged above.